### ISSUE TITLE
prognostic run: `total_precipitation` refactor in runfile loop

### ISSUE THREAD

**Issue by brianhenn** [VulcanClimateModeling/fv3net#1159]
The handling of the `total_precipitation` variable in the sklearn runfile loop is hard to follow if `total_precipitation` is one of the variables set/predicted in `_step_prephysics`. If computed in prephysics it's retained in the loop's `_state_updates` dict through the various steps until the `_apply_postphysics_to_dycore_state`, when it is applied as a state update. However, that step _also_ computes a `total_precipitation` as max(PRATE - <dQ2>, 0) in a separate dict of state updates. How these two are applied and which one is actually used (answer: the one from prephysics) is not obvious. 

Suggestions include refactoring the postphysics step's total precip computation to the ML and nudging steppers and making it optional in the steppers, and adding better logging of how `total_precipitation` is being computed.

**Comment by nbren12** [comment]
@brianhenn @AnnaKwa Is this still relevant now that 06934dc6494cd282cb404b3624c084292a1329f4 is in?

**Comment by AnnaKwa** [comment]
Yes, that commit only affected the ML stepper, which gets used in the postphysics update. The first issue described here, where a state update can be computed in the prephysics step but applied in the postphysics step, is still relevant. The second issue (unclear which total precip is used) has a comment line that explains what's going on (but the implementation is the same).
